mds/admin_overload.py

```diff
@@ -0,0 +1,86 @@
+from django.contrib import admin
+
+from uuid import UUID
+
+"""
+    to use when searching for devices in get_search_results as a relationship to self.model
+"""
+
+
+def get_devices_queryset_search_results(self, search_term):
+    def is_uuid(uuid_string, version=4):
+        try:
+            UUID(uuid_string, version=version)
+        except ValueError:
+            return False
+        return True
+
+    custom_queryset = self.model.objects.select_related("device__provider")
```

Comment:
This is a generic function, you can't impose that `select_related` to all models.